### DISCUSSION TITLE
Verifing cms signed data

### DISCUSSION THREAD

**Discussion by Dvergatal** [openssl/openssl#23498] [Q&A]
Hi all,
I'm trying to verify signed data file for that I'm using openssl 3.2.1 and the problem I'm occurring is that I do not know how properly to do that. First I will show my command for signing which is:
```bash
$ openssl cms -sign -in  sw-description -out sw-description.sig -signer swupdate.crt -inkey swupdate.key -outform DER -nosmimecap -binary
```

Now from what I have read in documentation it is supposed  to be something like this:
```bash
$ openssl cms -verify -binary -CAfile intermediate.cert.pem -inform der -certfile swupdate.crt -in sw-description.sig
```

but this is giving mi an error:
```bash
CMS Verification failure
4077377E707F0000:error:1700007F:CMS routines:check_content:no content:../openssl-3.2.1/crypto/cms/cms_smime.c:83:
```

Can anybody help me with this?

**Comment by JosephMarcia** [comment]
I encountered this issue, you can use "-content" and include the unsigned file in order to resolve this error. For example I created a signed .der file using a root CA I generated:

`openssl cms -sign -in signDer.der -signer root-cert.pem -inkey root-privkey.pem -out signed.cms -outform PEM -binary -md sha256`

Then to verify I used the following command:
`openssl cms -verify -in signed.cms -inform PEM -no_content_verify -noverify -content signDer.der -certfile root-cert.pem`

Openssl returns "CMS verification okay"

Because I used a self-signed root I had to include the -noverify option, but you shouldn't need it if you instead used a certificate that was signed by an CA.
